jpos/src/main/java/org/jpos/util/log/event/XMLLogEvent.java

```diff
@@ -0,0 +1,130 @@
+package org.jpos.util.log.event;
+
+import org.jdom2.Element;
+import org.jdom2.output.Format;
+import org.jdom2.output.XMLOutputter;
+import org.jpos.util.Loggeable;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.List;
+
+/**
+ * Created by erlangga on 2019-10-12.
+ */
+public class XMLLogEvent implements BaseLogEvent {
+
+    @Override
+    public String dumpHeader(PrintStream p, String indent, String realm, Instant dumpedAt, Instant createdAt, boolean noArmor) {
+        if (noArmor) {
+            p.println("");
+        } else {
+            if (dumpedAt == null)
+                dumpedAt = Instant.now();
+            StringBuilder sb = new StringBuilder(indent);
+            sb.append ("<log realm=\"");
+            sb.append (realm);
+            sb.append("\" at=\"");
+            sb.append(LocalDateTime.ofInstant(dumpedAt, ZoneId.systemDefault()));
+            sb.append ('"');
+            long elapsed = Duration.between(createdAt, dumpedAt).toMillis();
+            if (elapsed > 0) {
+                sb.append (" lifespan=\"");
+                sb.append (elapsed);
+                sb.append ("ms\"");
+            }
+            sb.append ('>');
+            p.println (sb.toString());
+        }
+        return indent + "  ";
+    }
+
+    @Override
+    public void dumpTrailer(PrintStream p, String indent, boolean noArmor) {
```

Comment:
same as above.

Reply:
It's legacy code, but it's used 

`p.println (indent + "</log>");`